### PR TITLE
Fix NPE when MP element can't be found in local asset manager store

### PR DESCRIPTION
**Pull request by mtneug**
Opencast currently assumes that assets can be found locally when checking for the correct local path. This is not true when using remote asset stores (e.g. S3), resulting in an NPE. This patch adds a null check.

### How to test this patch

1. Configure S3 asset store
2. Force all files to move to S3
3. Try to retrieve assets from asset manager endpoint

### Your pull request should…

* [x] have a concise title
* [ ] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)
* [ ] explain why it needs to be merged into the legacy branch, if it is targeting the legacy branch

### CODE REVIEW

File: modules/shared-filesystem-utils/src/main/java/org/opencastproject/assetmanager/util/AssetPathUtils.java
Comment (lkiesow) [inline modules/shared-filesystem-utils/src/main/java/org/opencastproject/assetmanager/util/AssetPathUtils.java]: This seems wrong and feels like this will surely have side effects. The function accepts a list of potential local paths, iterating over the list and filters them using some criteria. With your patch, the first entry just skips all checks and requirements, so that the results does no longer guarantee these properties.

I don't know this code well, but I would expect `null` to be a valid return value if there is no valid local path for the (remote) file.

Comment (mtneug) [reply modules/shared-filesystem-utils/src/main/java/org/opencastproject/assetmanager/util/AssetPathUtils.java]: `null` is not valid. Line 133 initializes the value to `null`. Lines 134-139, check each path in `localPaths` to see if the mediapackage folder exists in that path. The Line 140 assumes that the loop actually found a path, which is not a valid assumption in the case of remote asset managers. This line will throw an exception if `localPath` is `null`. Line 141 then again checks if the actual file exists. For this reason, it is ok to set the default `localPath` to the first one.

Comment (mtneug) [reply modules/shared-filesystem-utils/src/main/java/org/opencastproject/assetmanager/util/AssetPathUtils.java]: We discussed this in the dev meeting and agreed that this approach is confusing. I changed the path to a simple null check.

**Review by gregorydlogan — APPROVED** [review]

**Comment by lkiesow** [comment]
If you are waiting for my comment to merge this, I'm fine if you discussed this at the technical meeting. I don't need another look. I trust Greg.

**Pull request merged**
